### PR TITLE
Decouple './client/data/multi-currency'

### PR DESCRIPTION
**Pull request by lovo-h**
Fixes 106-gh-Automattic/woopayments.

## Changes proposed in this Pull Request

The changes in this PR starts the MCCY migration process to v2. Specifically, the MCCY data layer is migrated to v2. However, this is an intermediary change that will depend on the changes from the following issues to be completed before it is fully functional:
* 107-gh-Automattic/woopayments.
* 108-gh-Automattic/woopayments.

## Testing instructions

**Confirm Full Migration of `./client/data/multi-currency`**
* Start by running the `npm run test:js` command and ensure all tests pass.
* Ensure all files in `./client/data/multi-currency` were moved to `./multi-currency/client/data`, properly disconnected from WooPayments, properly reconnected into MCCY, and interfaced with WooPayments. Note: since there are two other GH issues that handle portions of the migration, there will be some MCCY files that will still need their MCCY data references migrated.
* Ensure WebPack was configured correctly by double-checking the following files: `./tsconfig.json`, `./tests/js/jest.config.js`, and `./webpack/shared.js`

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9314` or branch name `migrate/data-multi-currency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=migrate/data-multi-currency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9cc0432153d6f6145e5c31bc66ca8ca572566743
- Build time: 2024-08-29 17:17:07 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: webpack/shared.js [hunk 1]
```diff
@@ -113,9 +113,14 @@ module.exports = {
...
 		extensions: [ '.ts', '.tsx', '.json', '.js', '.jsx' ],
-		modules: [ path.join( process.cwd(), 'client' ), 'node_modules' ],
+		modules: [
+			path.join( process.cwd(), 'client' ),
+			path.join( process.cwd(), 'multi-currency', 'client' ),
+			'node_modules',
+		],
 		alias: {
 			assets: path.resolve( process.cwd(), 'assets' ),
+			mccy: path.resolve( process.cwd(), 'multi-currency', 'client' ),
```
Comment (bborman22) [inline webpack/shared.js]: As we discussed on the call, let's rename this to `multi-currency` in favor of readability over brevity.

File: multi-currency/client/data/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from './constants';
+import { initStore } from './store';
+
+initStore();
```
Comment (bborman22) [inline multi-currency/client/data/index.ts:9]: As discussed on the call, I tried loading the multi-currency settings page and got an error for the `wc/payments` store already being registered. This is likely due to what this PR is trying to accomplish by staying lean and avoiding merge conflicts. We decided a path forward would be to confirm the progress on the other two ongoing issues and if those are far enough along, we'll merge this PR as is and handle the full updated linking in the relevant PRs. If those issues are not far along, we'll update the linking in this PR (or just one of the others) to give us a better sense of where the gaps are now.

**Review by bborman22 — CHANGES_REQUESTED** [review]
As we discussed on the call I covered the following in my testing / review and left some comments on the open questions and requests:

- I confirmed that the JS tests pass
- I confirmed that the build scripts all worked and generated the expected assets
- Understanding that there wouldn't be full functionality, I still gave it a test run and see how things worked out. I found that the settings page in particular errors out and I'll leave more details on the comments on that.

The one confirmed change request is on the renaming of the alias to be more explicit. And one pending change request is whether we want / need to update the current linking to get this PR in a working state now or handle those updates in their relevant issues.

File: webpack/shared.js (hunk 1 above)
Comment (lovo-h) [reply webpack/shared.js]: Good call. Fixed in [1f01897](https://github.com/Automattic/woocommerce-payments/pull/9314/commits/1f018971ec4ce84254f458b827033c9b528ffff5).

File: multi-currency/client/data/index.ts (hunk 2 above)
Comment (lovo-h) [reply multi-currency/client/data/index.ts:9]: Correct. Here are links to the other two PRs for reference:
* #9326 
* #9348 

Those two PRs are far along already, so I will let each PR do their respective linking.

**Review by bborman22 — APPROVED** [review]
Thanks for making those changes and confirming the other updated linking will happen in their respective PRs.

I reviewed the changes again and confirmed JS tests are passing, it builds successfully, and that we use the full `mulit-currency` for the alias.

LGTM!

**Pull request merged**